src/modules/dashboard/components/Dialogs/EscrowFundsDialog/EscrowFundsDialogForm.tsx

```diff
@@ -0,0 +1,426 @@
+  },
+  partialFund: {
+    id: 'dashboard.EscrowFundsDialog.partialFund',
+    defaultMessage: '{name} to fund',
+  },
+  total: {
+    id: 'dashboard.EscrowFundsDialog.total',
+    defaultMessage: 'Total',
+  },
+  createDomain: {
+    id: 'dashboard.EscrowFundsDialog.creationTarget',
+    defaultMessage: 'Motion will be created in',
+  },
+});
+
+interface Props {
+  colony: Colony;
+  isForce: boolean;
+  setIsForce: React.Dispatch<React.SetStateAction<boolean>>;
+  onSubmit: () => void;
+}
+
+const EscrowFundsDialogForm = ({
+  colony,
+  isForce,
+  setIsForce,
+  onSubmit,
+}: Props) => {
+  const { values } = useFormikContext<FormValues>();
+  const [domainID, setDomainID] = useState<number>();
+  const { formatMessage } = useIntl();
+
+  useEffect(() => {
+    if (values.force !== isForce) {
+      setIsForce(values.force);
+    }
+  }, [isForce, setIsForce, values]);
+
+  const handleMotionDomainChange = useCallback(
+    (motionDomainId) => setDomainID(motionDomainId),
+    [],
+  );
+
+  const getDomainColor = useCallback<(domainId: string | undefined) => Color>(
+    (domainId) => {
+      const rootDomainColor: Color = Color.LightPink;
+      const defaultColor: Color = Color.Yellow;
+      if (domainId === String(ROOT_DOMAIN_ID)) {
+        return rootDomainColor;
+      }
+      if (!colony || !domainId) {
+        return defaultColor;
+      }
+      const domain = colony?.domains.find(
+        ({ ethDomainId }) => Number(domainId) === ethDomainId,
+      );
+      return domain ? domain.color : defaultColor;
+    },
+    [colony],
+  );
+
+  const renderActiveOption = useCallback<
+    (option: SelectOption | undefined, label: string) => ReactNode
+  >(
+    (option, label) => {
+      const value = option?.value;
+      const color = getDomainColor(value);
+
+      return (
+        <div className={styles.activeItem}>
+          <ColorTag color={color} />
+          <div className={styles.activeItemLabel}>{label}</div>
+        </div>
+      );
+    },
+    [getDomainColor],
+  );
+
+  const renderActiveDomainOption = useCallback<
+    (option: SelectOption | undefined, label: string) => ReactNode
+  >(
+    (option, label) => {
+      let displayLabel =
+        parseInt(option?.value || `${ROOT_DOMAIN_ID}`, 10) === ROOT_DOMAIN_ID
+          ? `${formatMessage(MSG.createDomain)} ${label}`
+          : `${formatMessage(MSG.createDomain)} ${formatMessage({
+              id: 'domain.root',
+            })}/${label}`;
+
+      if (!option) {
+        displayLabel = `${formatMessage({ id: 'domain.root' })}`;
+      }
+      return <div className={styles.motionActiveItem}>{displayLabel}</div>;
+    },
+    [formatMessage],
+  );
+
+  const filterDomains = useCallback((optionDomain) => {
+    const optionDomainId = parseInt(optionDomain.value, 10);
+
+    return optionDomainId !== COLONY_TOTAL_BALANCE_DOMAIN_ID;
+  }, []);
+
+  // @TODO replace the mock with real data
+  const balanceOptions = useMemo(
```

Comment:
To be honest, I don't understand why you need to write `balanceOptions` & `requiredFunds ` as separate useMemo parts, I think it can be written in the code and will be easier to read...